mysql-test/suite/innodb/t/avoid_deadlock_with_blocked.test

```diff
@@ -0,0 +1,229 @@
+--source include/have_innodb.inc
+--source include/have_debug_sync.inc
```

Comment:
It would be good to check for `have_debug.inc` as well, to ensure that the test is skipped at the start of an `mtr` run, rather than in the middle, of a `CMAKE_BUILD_TYPE≠Debug` build.